tests/test_adjoint_jacobian.py

```diff
@@ -1079,3 +1105,41 @@ def circuit(params):
     j_cpu = qml.jacobian(qnode_cpu)(params)
 
     assert np.allclose(j_cpu, j_gpu)
+
+
+@pytest.mark.parametrize(
+    "obs,obs_type",
+    [
+        (qml.PauliZ(0), NamedObsGPU_C64),
+        (qml.PauliZ(0) @ qml.PauliZ(1), TensorProdObsGPU_C64),
+        (qml.Hadamard(0), NamedObsGPU_C64),
+        (qml.Hamiltonian([1], [qml.PauliZ(0)]), HamiltonianGPU_C64),
+        (
+            qml.PauliZ(0) @ qml.Hadamard(1) @ (0.1 * (qml.PauliZ(2) + qml.PauliX(3))),
+            TensorProdObsGPU_C64,
+        ),
+        (
+            qml.SparseHamiltonian(qml.Hamiltonian([1], [qml.PauliZ(0)]).sparse_matrix(), wires=[0]),
+            SparseHamiltonianGPU_C64,
+        ),
+    ],
+)
+def test_obs_returns_expected_type(obs, obs_type):
+    """Tests that observables get serialized to the expected type."""
+    assert isinstance(plk._serialize._serialize_ob(obs, dict(enumerate(obs.wires)), True), obs_type)
```

Comment:
Hi, @timmysilv. This `plk` is coming from the suite of tests for PennyLane Lightning Kokkos.
This (and all others) will need to be updated for this repository.

Reply:
Looks like @multiphaseCFD is implementing the necessary changes. Forget about this comment.

Reply:
oops sorry about that! I swear I had fixed the copy-paste locally but it's been lost. ty for the fix!